### DISCUSSION TITLE
Two keys appear at the same time

### DISCUSSION THREAD

**Discussion by Liuccysdgg** [nlohmann/json#2667] [Q&A] [answered]
The following JSON string will appear with no error：

{
“name”:null,
"name":"lili"
}

This is generated by nlohmann/json, because reading before writing will generate this kind of JSON。

**Comment by nlohmann** [comment, marked as answer]
Duplicate object keys are not an error in JSON.

From RFC 8259:

> An object whose names are all unique is interoperable in the sense
> that all software implementations receiving that object will agree on
> the name-value mappings.  When the names within an object are not
> unique, the behavior of software that receives such an object is
> unpredictable.  Many implementations report the last name/value pair
> only.  Other implementations report an error or fail to parse the
> object, and some implementations report all of the name/value pairs,
> including duplicates.

This library stores the last seen key.

**Comment by Liuccysdgg** [comment]
Thank you very much for your reply. This is my negligence. I always thought that the JSON key is unique and does not repeat

发送自 Windows 10 版邮件<https://go.microsoft.com/fwlink/?LinkId=550986>应用

发件人: Niels Lohmann<mailto:notifications@github.com>
发送时间: 2021年3月5日 21:28
收件人: nlohmann/json<mailto:json@noreply.github.com>
抄送: uTigger<mailto:a1585346868@hotmail.com>; Author<mailto:author@noreply.github.com>
主题: Re: [nlohmann/json] Two keys appear at the same time (#2667)


Duplicate object keys are not an error in JSON.

From RFC 8259:

An object whose names are all unique is interoperable in the sense
that all software implementations receiving that object will agree on
the name-value mappings. When the names within an object are not
unique, the behavior of software that receives such an object is
unpredictable. Many implementations report the last name/value pair
only. Other implementations report an error or fail to parse the
object, and some implementations report all of the name/value pairs,
including duplicates.

This library stores the last seen key.

―
